### PR TITLE
fix: move to using insecure grpc channels with emulator

### PR DESCRIPTION
**Pull request by crwilcox**
Related to #359

On windows, configuring a secure channel with credentials will not connect.

### CODE REVIEW

File: google/cloud/firestore_v1/base_client.py [hunk 1]
```diff
@@ -177,15 +177,20 @@ def _emulator_channel(self, transport):
...
         # "transport.create_channel" to create gRPC channels once google-auth
         # extends it's allowed credentials types.
+        # if "GrpcAsyncIOTransport" in str(transport.__name__):
+        #     return grpc.aio.secure_channel(
+        #         self._emulator_host, self._local_composite_credentials()
+        #     )
+        # else:
+        #     return grpc.secure_channel(
+        #         self._emulator_host, self._local_composite_credentials()
+        #     )
```
Comment (tseaver) [inline google/cloud/firestore_v1/base_client.py:187]: Do we really want to leave this commented-out bit in place?

Comment (crwilcox) [reply google/cloud/firestore_v1/base_client.py:187]: no ;) but this is mostly here to spark conversation. I am not sure we can even make this change without regressing something.

File: google/cloud/firestore_v1/base_client.py [hunk 2]
```diff
@@ -178,13 +178,9 @@ def _emulator_channel(self, transport):
         # "transport.create_channel" to create gRPC channels once google-auth
         # extends it's allowed credentials types.
         if "GrpcAsyncIOTransport" in str(transport.__name__):
-            return grpc.aio.secure_channel(
-                self._emulator_host, self._local_composite_credentials()
-            )
+            return grpc.aio.insecure_channel(self._emulator_host)
```
Comment (craiglabenz) [inline google/cloud/firestore_v1/base_client.py:181]: Can we possibly have our cake and eat it too by introducing another environment variable for these custom credentials?

**Comment by crwilcox** [comment]
@paulharter FYI. This should still support your use case, but as there was no test added to demonstrate your complete use, I am unsure. Reach out if this doesn't work and we can find a way to support both use case :)

File: google/cloud/firestore_v1/base_client.py [hunk 3]
```diff
@@ -167,50 +167,25 @@ def _firestore_api_helper(self, transport, client_class, client_module) -> Any:
 
     def _emulator_channel(self, transport):
         """
-        Creates a channel using self._credentials in a similar way to grpc.secure_channel but
-        using grpc.local_channel_credentials() rather than grpc.ssh_channel_credentials() to allow easy connection
-        to a local firestore emulator. This allows local testing of firestore rules if the credentials have been
-        created from a signed custom token.
+        Creates an insecure channel to communicate with the local emulator.
+        If credentials are provided the token is extracted and added to the
+        headers. this supports local testing of firestore rules if the credentials
```
Comment (craiglabenz) [inline google/cloud/firestore_v1/base_client.py:172]: nit: `This supports` (capitalization)

File: google/cloud/firestore_v1/base_client.py [hunk 4]
```diff
@@ -167,50 +167,25 @@ def _firestore_api_helper(self, transport, client_class, client_module) -> Any:
...
 
         :return: grpc.Channel or grpc.aio.Channel
         """
-        # TODO: Implement a special credentials type for emulator and use
-        # "transport.create_channel" to create gRPC channels once google-auth
-        # extends it's allowed credentials types.
+        # Insecure channels are used for the emulator as secure channels
+        # cannot be used to communicate on some environments.
+        # https://github.com/googleapis/python-firestore/issues/359
+        token = "owner"
```
Comment (craiglabenz) [inline google/cloud/firestore_v1/base_client.py:180]: A quick comment explaining this default might help posterity

File: tests/unit/v1/test_base_client.py [hunk 5]
```diff
@@ -160,21 +160,23 @@ def test_emulator_channel(self):
...
             )
-        )
-        self.assertTrue(
-            isinstance(
-                composite_credentials._credentials._channel_credentials,
-                grpc._cython.cygrpc.LocalChannelCredentials,
+        with mock.patch("grpc.insecure_channel") as insecure_channel:
+            channel = client._emulator_channel(FirestoreGrpcTransport)
+            insecure_channel.assert_called_once_with(
+                emulator_host, options=[("Authorization", "Bearer test")]
```
Comment (craiglabenz) [inline tests/unit/v1/test_base_client.py:178]: Nice test :)

**Review by craiglabenz — APPROVED** [review]
LGTM

**Pull request merged**